internal/adapters/clients/ollama.go

```diff
@@ -11,110 +11,23 @@ import (
-	if err != nil {
-		return openai.ChatCompletionResponse{}, fmt.Errorf("failed to create request: %v", err)
-	}
-
-	req.Header.Set("Content-Type", "application/json")
-
-	resp, err := c.cfg.HTTPClient.Do(req)
-	if err != nil {
-		return openai.ChatCompletionResponse{}, fmt.Errorf("failed to send request: %v", err)
-	}
-	defer resp.Body.Close()
-
-	respBody, err := io.ReadAll(resp.Body)
-	if err != nil {
-		return openai.ChatCompletionResponse{}, fmt.Errorf("failed to read response: %v", err)
-	}
-
-	var ollamaResponse ollamaResponse
-
-	err = json.Unmarshal(respBody, &ollamaResponse)
-	if err != nil {
-		return openai.ChatCompletionResponse{}, fmt.Errorf("failed to unmarshal response: %v", err)
-	}
-
-	// convert ollamaResponse to openai.ChatCompletionResponse
-	return convertOllamaRespToOpenAIResponse(ollamaResponse), nil
-}
+// CreateChatCompletion sends a request to the Ollama server and returns the response.
+func (c *OllamaClient) CreateChatCompletion(ctx context.Context, request openai.ChatCompletionRequest) (response openai.ChatCompletionResponse, err error) {...}
 
-type ollamaRequest struct {
-	Model    string          `json:"model"`
-	Messages []ollamaMessage `json:"messages"`
-	Stream   bool            `json:"stream"`
-}
-type ollamaMessage struct {
-	Role    string `json:"role"`
-	Content string `json:"content"`
-}
-
-type ollamaResponse struct {
-	Model              string        `json:"model"`
-	CreatedAt          string        `json:"created_at"`
-	Message            ollamaMessage `json:"message"`
-	Done               bool          `json:"done"`
-	TotalDuration      float64       `json:"total_duration"`
-	PromptEvalCount    int           `json:"prompt_eval_count"`
-	PromptEvalDuration float64       `json:"prompt_eval_duration"`
-	EvalCount          int           `json:"eval_count"`
-	EvalDuration       float64       `json:"eval_duration"`
-}
-
-func convertToOllamaRequest(request openai.ChatCompletionRequest) ollamaRequest {
-	ollamaRequest := ollamaRequest{
-		Model:    request.Model,
-		Messages: []ollamaMessage{},
-		Stream:   request.Stream,
-	}
-
-	for _, message := range request.Messages {
-		ollamaMessage := ollamaMessage{
-			Role:    message.Role,
-			Content: message.Content,
-		}
+// Converts an OpenAI request format to Ollama's required format.
+func convertToOllamaRequest(request openai.ChatCompletionRequest) ollamaRequest {...}
 
-		ollamaRequest.Messages = append(ollamaRequest.Messages, ollamaMessage)
-	}
-
-	return ollamaRequest
-}
-
-func convertOllamaRespToOpenAIResponse(response ollamaResponse) openai.ChatCompletionResponse {
-	return openai.ChatCompletionResponse{
-		Model: response.Model,
-		Choices: []openai.ChatCompletionChoice{
-			{
-				Message: openai.ChatCompletionMessage{
-					Role:    response.Message.Role,
-					Content: response.Message.Content,
-				},
-			},
-		},
-	}
-}
+// Converts Ollama's response back into OpenAI's response format.
```

Comment:
Implementation details should be added for 'convertToOllamaRequest'. Documentation could also be helpful explaining how it transforms the request.